src/products/components/Products.vue

```diff
@@ -0,0 +1,154 @@
+<template>
+  <div>
+    <v-data-table
+      :headers="headers"
+      :items="products"
+      class="mt-10"
+      :item-class="itemRowBackground"
+      show-select
+      item-key="productNumber"
+      disable-pagination
+      hide-default-footer
+      :search="search"
+      dense
+      @click:row="navigateProduct"
+    >
+      <template v-slot:[`item.status`]="{ item }">
+        <td class="d-flex">
+          <v-switch
+            v-model="item.status"
+            :label="item.status | status"
+            @click.capture.native.stop="confirmChangeStatus(item)"
+          >
+          </v-switch>
+        </td>
+      </template>
+      <template v-slot:[`item.productPrices[0].price`]="{ item }">
+        <td class="d-flex j-content-space-between">
+          <span class="w-100per t-align-right mr-1"
+            ><b>{{ item.productPrices[0].price | number }}</b></span
+          >
+          <span>{{ item.productPrices[0].currency | upperCase }}</span>
+        </td>
+      </template>
+      <template v-slot:[`item.actions`]="{ item }">
+        <v-menu offset-y>
+          <template v-slot:activator="{ on }">
+            <v-btn v-on="on" icon>
+              <v-icon>mdi-dots-vertical</v-icon>
+            </v-btn>
+          </template>
+          <v-list>
+            <v-list-item
+              v-if="item.status === 0"
+              @click="confirmChangeStatus(item)"
+            >
+              Enable
+            </v-list-item>
+            <v-list-item
+              v-else
+              @click="confirmChangeStatus(item)"
+              :rules="item.stattus === 0"
+            >
+              Disable
+            </v-list-item>
+            <v-list-item @click="confirmDeleteProduct(item)">
+              Delete
+            </v-list-item>
+          </v-list>
+        </v-menu>
+      </template>
+    </v-data-table>
+    <Dialog
+      :title="dialogTitle"
+      :content="dialogContent"
+      :isShowDialog="isShowDialog"
+      @setShowDialog="setShowDialog"
+      :btnAccept="'OK'"
+      :btnNotAccept="'Cancel'"
+      @confirmChange="
+        actionType === 'changeStatus' ? changeStatus() : deleteProduct()
+      "
+    ></Dialog>
+  </div>
+</template>
+
+<script>
+import Dialog from "@/dialogs/views/Dialog.vue";
+import "@/shared/style/style.css";
+export default {
+  components: {
+    Dialog,
+  },
+  props: {
+    products: [],
+    search: String,
+  },
+  data() {
+    return {
+      headers: [
+        {
+          text: "NUMBER",
+          align: "start",
+          sortable: true,
+          value: "productNumber",
+        },
+        { text: "NAME", value: "name", sortable: false },
+        { text: "STATUS", value: "status", sortable: false },
+        { text: "PRICE", value: "productPrices[0].price", sortable: false },
+        { text: "", value: "actions", sortable: false },
+      ],
+      productChange: {},
```

Comment:
The term "selectedProduct" will be more precise.
and init state should be `undefined`
```suggestion
      selectedProduct: undefined,
```